src/Apps/Auction/Components/BidForm.tsx

```diff
@@ -57,18 +63,38 @@ const getSelectedBid = ({
   return selectedIncrement.value
 }
 
+const determineDisplayRequirements = (
+  bidder: BidForm_saleArtwork["sale"]["registrationStatus"],
+  me: BidForm_me
+) => {
+  const isRegistered = !!bidder
+
+  return {
+    requiresCheckbox: !isRegistered,
+    requiresPaymentInformation: !(isRegistered || me.hasQualifiedCreditCards),
+  }
+}
```

Comment:
Nice, I like capturing these points in one area.

Reply:
+1 – really readable code!